cmd/internal/planetscale_edge_database.go

```diff
@@ -285,50 +281,99 @@ func (p PlanetScaleEdgeDatabase) sync(ctx context.Context, tc *psdbconnect.Table
 		keyspaceOrDatabase = ps.Database
 	}
 
-	// Stop when we've reached the well known stop position for this sync session.
-	watchForVgGtidChange := false
+	// Can finish sync once we've synced to the stop position, or finished the VStream COPY phase
+	canFinishSync := false
 	resultCount := 0
 
+	var fields []*query.Field
+
 	for {
 		res, err := c.Recv()
 		if err != nil {
-			p.Logger.Log(LOGLEVEL_INFO, fmt.Sprintf("%sExiting sync and flushing records due to error: %+v", preamble, err))
+			if s, ok := status.FromError(err); ok && s.Code() == codes.DeadlineExceeded && canFinishSync {
+				// No next VGTID found, but we previously found stop position or finished VStream COPY phase
+				p.Logger.Log(LOGLEVEL_INFO, fmt.Sprintf("%sExiting sync and flushing records because no new VGTID found after stop position %+v", preamble, stopPosition))
```

Comment:
FYI, this is the line that explicitly handles:
> Explicitly handle "no new VGTIDs". This can happen if there was no new activity on the database after the stop position.